Http/Controllers/API/V1/PriceController.php

```diff
@@ -23,75 +24,77 @@ public function index(Request $request)
             'query' => 'nullable|string|max:50',
         ]);
 
-        $lists = [];
-        if ($request->has('list_id')) {
-            $lists[] = $request->get('list_id');
-        } else if ($request->has('list_ids')) {
-            $lists = $request->array('list_ids');
-        } else if ($request->has('reimbursement_type_id')) {
-            $query = TenantPriceList::forReimbursementType($request->get('reimbursement_type_id'))
-                ->used();
+        $lists = $this->buildListsArrayFromRequestParams($request);
+        $authIssueResponse = $this->authorizationCheckForLists($lists);
+        if ($authIssueResponse) {
+            return $authIssueResponse;
+        }
 
-            if ($priceList = $query->first()) {
-                $lists[] = $priceList;
-            }
+        $query  = PriceQueryHelper::buildPriceOnlyQuery($request,$lists);
+        if(empty($query)) return $this->emptyItem();
+
+        if ($request->has('query')) {
+           $query = PriceQueryHelper::applyQueryParamFromRequest($request,$query);
         }
 
-        if (count($lists)) {
-            foreach ($lists as $key => $list) {
-                $list = is_numeric($list) ? TenantPriceList::find($list) : $list;
-                $this->authorize('view', [$list, TenantPriceList::class]);
-                $lists[$key] = $list->id;
-            }
-        } else {
-            $this->authorize('view', [null, TenantPriceList::class]);
+        if ($request->has('sort_by')) {
```

Comment:
This is potentially conflicting with sorting logic according to relevance with is now outsourced in the helper class. Not sure if any of them has higher priority, but as it was sequentially applied in the original file, I'm keeping it the same. Otherwise, this sort could potentially be moved to PriceQueryHelper::applyQueryParamFromRequests and maybe some of the sorting eliminated.